src/Textarea.jsx

```diff
@@ -137,15 +155,20 @@ class ReactTextareaAutocomplete extends React.Component {
         this.textareaRef.dispatchEvent(e);
         if (onChange) onChange(e);
 
-        this.setTextareaCaret(newCaretPosition);
+        this.setCaretPosition(newCaretPosition);
+
+        if (onCaretPositionChange) {
```

Comment:
This will trigger the callback only when user confirm his choice, right? It would be better to attach this for "keyPress" and "mouseClick" event, to handle all cases of caret change, what do you think?